api/config/esgateway/config_request.go

```diff
@@ -89,6 +91,20 @@ func (c *ConfigRequest) Validate() error {
 // to start the service.
 func (c *ConfigRequest) PrepareSystemConfig(creds *ac.TLSCredentials) (ac.PreparedSystemConfig, error) {
 	c.V1.Sys.Tls = creds
+
+	enableSystemNameServer := c.GetV1().GetSys().GetNgx().GetMain().GetResolvers().GetEnableSystemNameservers().GetValue()
+	nameServers := c.GetV1().GetSys().GetNgx().GetMain().GetResolvers().GetNameservers()
+	if len(nameServers) == 0 && enableSystemNameServer {
+		c.V1.Sys.Ngx.Main.Resolvers.NameserversString = getSystemResolvers()
+	} else {
+		if len(nameServers) > 0 {
+			ns := make([]string, 0, len(nameServers))
+			for _, n := range nameServers {
+				ns = append(ns, n.GetValue())
```

Comment:
[nit] maybe check `n != nil`. I think the worst thing that will happen in this case is there will be extra spaces in the namservers string